### PR TITLE
Doc: Add note and example for date math in conditionals

### PR DESCRIPTION
**Pull request by karenzone**
## Release notes
[rn:skip] 

## What does this PR do?
Updates docs with info about date math in conditionals and provides a workaround example.
Brings topic up to current documentation standards for heading capitalization and tagging

**PREVIEW:** https://logstash_13199.docs-preview.app.elstc.co/guide/en/logstash/master/event-dependent-configuration.html#conditionals

### CODE REVIEW

File: docs/static/configuration.asciidoc [hunk 1]
```diff
@@ -510,18 +510,45 @@ doesn't exist versus a field that's simply false. The expression `if [foo]` retu
 
 For more complex examples, see <<using-conditionals, Using Conditionals>>.
 
-[float]
+[discrete]
```
Comment (robbavey) [inline docs/static/configuration.asciidoc:513]: I like the content, but I think positionally, we might want to put this under the `@metadata` section - the workaround uses the `@metadata` field, and the context could be useful. Also, I suspect that this topic is _probably_  less relevant to more users than the `@metadata` topic

File: docs/static/configuration.asciidoc [hunk 2]
```diff
@@ -510,18 +510,45 @@ doesn't exist versus a field that's simply false. The expression `if [foo]` retu
...
+===== Date math in conditionals
+
+Date math in conditionals is not currently supported, but a workaround is available. 
+Put the date calculation in a field so that you can use the field reference in a conditional. 
+
+*Example* 
+
+This statement won't work: 
+
+`if "%{+HH}:%{+mm}" < "15:30" {`
```
Comment (robbavey) [inline docs/static/configuration.asciidoc:524]: Maybe add the 

`add_field => {"string_compare" => "%{+HH}:%{+mm} is before 15:30"}`

part to this the broken example to give an equivalence between the working and non-working versions?

File: docs/static/configuration.asciidoc [hunk 3]
```diff
@@ -510,18 +510,45 @@ doesn't exist versus a field that's simply false. The expression `if [foo]` retu
 
 For more complex examples, see <<using-conditionals, Using Conditionals>>.
 
-[float]
+[discrete]
+[[date-math]]
+===== Date math in conditionals
+
+Date math in conditionals is not currently supported, but a workaround is available.
```
Comment (robbavey) [inline docs/static/configuration.asciidoc:517]: If we put this content below `@metadata`, maybe we can refer to it here, stating something like "a workaround using the `@metadata` field is available"

Unless that is a bit long winded and unnecessary detail

**Review by robbavey — COMMENTED** [review]
I like the comment, couple of questions on content and positioning of this section

**Comment by karenzone** [comment]
Good review comments, @robbavey. They're in.

File: docs/static/configuration.asciidoc [hunk 4]
```diff
@@ -642,8 +647,7 @@ Pipeline main started
 That's it!  No extra fields in the output, and a cleaner config file because you
 do not have to delete a "timestamp" field after conversion in the `date` filter.
 
-Another use case is the CouchDB Changes input plugin (See
-https://github.com/logstash-plugins/logstash-input-couchdb_changes).
+Another use case is the https://github.com/logstash-plugins/logstash-input-couchdb_changesCouchDB[Changes input plugin].
```
Comment (robbavey) [inline docs/static/configuration.asciidoc]: Link to plugin is broken

File: docs/static/configuration.asciidoc [hunk 5]
```diff
@@ -663,8 +667,39 @@ output {
 }
 ----------------------------------
 
+[discrete]
+[[date-math]]
+===== Date math in conditionals
```
Comment (robbavey) [inline docs/static/configuration.asciidoc:672]: Maybe this should be `sprintf time format`?

File: docs/static/configuration.asciidoc [hunk 6]
```diff
@@ -510,18 +510,23 @@ doesn't exist versus a field that's simply false. The expression `if [foo]` retu
 
 For more complex examples, see <<using-conditionals, Using Conditionals>>.
 
-[float]
+NOTE: Date math in conditionals is not currently supported.
```
Comment (robbavey) [inline docs/static/configuration.asciidoc:513]: Maybe more appropriate as `sprintf time format` - `Date math` is more typically an Elasticsearch term

File: docs/static/configuration.asciidoc [hunk 7]
```diff
@@ -663,8 +667,39 @@ output {
...
+[discrete]
+[[date-math]]
+===== Date math in conditionals
+
+Date math in conditionals is not currently supported, but a workaround is available. 
+Put the date calculation in a field so that you can use the field reference in a conditional. 
+
+*Example* 
+
+This statement won't work:
```
Comment (robbavey) [inline docs/static/configuration.asciidoc:679]: I think a little more context would help here:

Something like:

This attempt to use add a field depending on the ingestion time won't work using `sprintf time format` directly

File: docs/static/configuration.asciidoc [hunk 8]
```diff
@@ -663,8 +667,39 @@ output {
...
+  add_field => {"string_compare" => "%{+HH}:%{+mm} is before 15:30"}
+```
+
+This workaround gives you the intended results:
+
+[source,js]
+----------------------------------
+mutate {
+  add_field => {"[@metadata][short_time]" => "%{+HH}:%{+mm}"}
+  }
```
Comment (robbavey) [inline docs/static/configuration.asciidoc:693]: Nit: formatting

File: docs/static/configuration.asciidoc [hunk 9]
```diff
@@ -663,8 +667,39 @@ output {
...
+if [@metadata][short_time] < "15:30" {
+  mutate {
+    add_field => {"string_compare" => "%{+HH}:%{+mm} is before 15:30"}
+    }
+    }
```
Comment (robbavey) [inline docs/static/configuration.asciidoc:698]: Nit: formatting

**Review by robbavey — CHANGES_REQUESTED** [review]
Couple of formatting nitpicks.

**Review by robbavey — APPROVED** [review]
LGTM

**Pull request merged**